src/Persistence_matrix/include/gudhi/Persistence_matrix/base_matrix_with_column_compression.h

```diff
@@ -344,11 +349,10 @@ class Base_matrix_with_column_compression : protected Master_matrix::Matrix_row_
   index nextColumnIndex_;                             /**< Next unused column index. */
   Column_settings* colSettings_;                      /**< Cell factory. */
   /**
-   * @brief Column factory.
-   * @warning As the member is static, they can eventually be problems if the matrix is duplicated in several threads.
-   * If this become necessary, the static should be removed in the future.
+   * @brief Column factory. Has to be a pointer as Simple_object_pool is not swappable, so their adresses have to be
+   * exchanged instead.
```

Comment:
Does this only come up in the compressed matrix? I think you are using pools elsewhere. I didn't check what those other uses of pools are like, could you tell me why they don't have this issue with moving/swapping?

Reply:
Ah, no, it is also indirectly behind a pointer for others
```
  Column_settings* colSettings_;  //pointer because the of swap operator on matrix_ which also stores the pointer
```